src/EventStore.php

```diff
@@ -388,4 +352,108 @@ public function getActionEventEmitter(): ActionEventEmitter
     {
         return $this->actionEventEmitter;
     }
+
+    private function preLoad(string $action, StreamName $streamName, int $fromNumber, int $count = null)
+    {
+        $argv = [
+            'streamName' => $streamName,
+            'fromNumber' => $fromNumber,
+            'count'      => $count,
+        ];
+
+        $event = $this->actionEventEmitter->getNewActionEvent($action, $this, $argv);
+
+        $this->getActionEventEmitter()->dispatch($event);
+
+        if ($event->propagationIsStopped()) {
+            $stream = $event->getParam('stream', false);
+
+            if ($stream instanceof Stream && $stream->streamName()->toString() == $streamName->toString()) {
+                return $stream;
+            }
+
+            throw StreamNotFoundException::with($streamName);
+        }
+
+        $streamName = $event->getParam('streamName');
+        $fromNumber = $event->getParam('fromNumber');
+        $count      = $event->getParam('count');
+
+        return [
+            $streamName,
+            $fromNumber,
+            $count,
+            $event
+        ];
+    }
+
+    private function postLoad(string $action, ?Stream $stream, StreamName $streamName, ActionEvent $event): ?Stream
+    {
+        if (! $stream) {
+            throw StreamNotFoundException::with($streamName);
+        }
+
+        $event->setName($action);
+
+        $event->setParam('stream', $stream);
+
+        $this->getActionEventEmitter()->dispatch($event);
+
+        if ($event->propagationIsStopped()) {
+            throw StreamNotFoundException::with($streamName);
+        }
+
+        return $event->getParam('stream');
+    }
+
+    private function preloadEventsByMetadataFrom(
```

Comment:
It returns Iterator|array. That's why there is no return type here. Also it's a private method, so nobody has to deal with it.